rllib/agents/ppo/ppo.py

```diff
@@ -410,6 +426,83 @@ def get_default_policy_class(self, config: TrainerConfigDict) -> Type[Policy]:
         else:
             return PPOTFPolicy
 
+    @ExperimentalAPI
+    def training_iteration(self) -> ResultDict:
+        # Collect SampleBatches from sample workers until we have a full batch.
+        if self._by_agent_steps:
+            train_batch = synchronous_parallel_sample(
+                worker_set=self.workers, max_agent_steps=self.config["train_batch_size"]
```

Comment:
max_env_steps and max_env_steps should be used more often in other training iteration functions where we always wait until we have _at least_ enough samples to continue and therefore sometimes don't use exact batch or samples sizes!

Reply:
Yes, totally agree, we should now unify this and remove the `while` loops everywhere else. We'll clean this up now that we are getting very close to setting `_disable_execution_plan_api=True` by default.